src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,114 @@
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
+interface Props {
+  todosFromServer: Todo[];
+  selectUser: React.Dispatch<React.SetStateAction<number>>;
+}
+
+export const TodoList: React.FC<Props>
+  = ({ todosFromServer, selectUser }) => {
+    const [filter, setFilter] = useState<string>('');
+    const [selectFilter, setSelectFilter] = useState('all');
+
+    const filteredByState
+      = todosFromServer.filter(todo => {
+        if (todo.completed === true && selectFilter === 'completed') {
+          return true;
+        }
+
+        if (todo.completed === false && selectFilter === 'active') {
+          return true;
+        }
+
+        if (selectFilter === 'all') {
+          return true;
+        }
+
+        return false;
+      });
+
+    const filteredByTitle
+      = filteredByState.filter(todo => todo.title.includes(filter));
+
+    return (
+      <div className="TodoList">
+        <h2>Todos:</h2>
+
+        <input
+          className="input"
+          type="text"
+          data-cy="filterByTitle"
+          value={filter}
+          onChange={event => {
+            setFilter(event.target.value);
+          }}
+        />
+
+        <div className="select">
+          <select
+            value={selectFilter}
+            onChange={event => {
+              setSelectFilter(event.target.value);
+            }}
           >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
+            <option value="all">all</option>
+            <option value="active">active</option>
+            <option value="completed">completed</option>
+          </select>
+
+        </div>
+
+        <button
+          type="button"
+          className="button"
+        >
+          Randomize
+        </button>
+
+        <div className="TodoList__list-container">
+          <ul
+            className="TodoList__list"
+            data-cy="listOfTodos"
           >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+            {filteredByTitle.map(todo => (
+              <li
+              // className={`TodoList__item TodoList__item--${todo.completed ? 'checked' : 'unchecked'}`}
+              // how can I make it using classNames ? I mean now here have duplicate.
```

Comment:
try something like this with classnames
```
  'TodoList__item--unchecked': !todo.completed,
  'TodoList__item--checked': todo.completed,
```

Reply:
![image](https://user-images.githubusercontent.com/63620508/174677447-b49fd786-0b2e-4015-b5be-aea26e4c4db1.png)

Yes, its already same in my code but I mean that in classNames I need two times writing 'TodoList__item--' but only ending need change.
In ternar operator its seems like less code ? Or maybe its need to doo just for more understanding code ?

Reply:
In this case, we have almost the same class name `todolist__item-but these are completely different classes, so it is better to separate the logic for them